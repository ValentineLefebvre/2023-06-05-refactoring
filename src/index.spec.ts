import runGoldenMaster from "jest-golden-master";
import { Game } from ".";

test("My first scenario", async () => {
	runGoldenMaster(async () => {
		const game = new Game();
		game.add("Mathieu");
		game.add("Thomas");
		game.add("Clément");
		game.roll(1);
		game.wasCorrectlyAnswered();
		game.roll(1);
		game.wasCorrectlyAnswered();
		game.roll(1);
		game.wasCorrectlyAnswered();
		game.roll(1);
		game.wasCorrectlyAnswered();
		game.roll(1);
		game.wasCorrectlyAnswered();
		game.roll(1);
		game.wasCorrectlyAnswered();
		game.roll(1);
		game.wasCorrectlyAnswered();
		game.roll(1);
		game.wasCorrectlyAnswered();
	});
});

test("2nd scenario : player goes to penalty box", async () => {
	runGoldenMaster(async () => {
		const game = new Game();
		game.add("Mathieu");
		game.add("Thomas");
		game.roll(1);
		game.wrongAnswer();
	});
});

test("3rd scenario : player goes to penalty box then comes out", async () => {
	runGoldenMaster(async () => {
		const game = new Game();
		game.add("Mathieu");
		game.add("Thomas");
		game.roll(1);
		game.wrongAnswer();
		game.roll(2);
		game.wasCorrectlyAnswered();
		game.roll(3);
		game.wasCorrectlyAnswered();
	});
});

test("4th scenario : player goes to penalty box and stays there", async () => {
	runGoldenMaster(async () => {
		const game = new Game();
		game.add("Mathieu");
		game.add("Thomas");
		game.roll(1);
		game.wrongAnswer();
		game.roll(2);
		game.wasCorrectlyAnswered();
		game.roll(4);
	});
});

test("5th scenario : player goes to penalty box, comes out but goes back into it", async () => {
	runGoldenMaster(async () => {
		const game = new Game();
		game.add("Mathieu");
		game.add("Thomas");
		game.roll(1);
		game.wrongAnswer();
		game.roll(2);
		game.wasCorrectlyAnswered();
		game.roll(3);
		game.wrongAnswer();
	});
});

test("6th scenario : one player circles the board", async () => {
	runGoldenMaster(async () => {
		const game = new Game();
		game.add("Mathieu");
		game.add("Thomas");
		game.add("Clément");
		game.roll(6);
		game.wasCorrectlyAnswered();
		game.roll(1);
		game.wasCorrectlyAnswered();
		game.roll(2);
		game.wasCorrectlyAnswered();
		game.roll(6);
		game.wasCorrectlyAnswered();
	});
});