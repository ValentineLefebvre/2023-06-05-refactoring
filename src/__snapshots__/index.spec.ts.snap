// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2nd scenario : player goes to penalty box 1`] = `
[
  "New player added : Mathieu at place 0, with 0 coin and not in penalty box.",
  "New player added : Thomas at place 0, with 0 coin and not in penalty box.",
  "Player Mathieu got 1 on their dice.",
  "Player Mathieu's new place is 1.",
  "Science Question 0",
  "Player Mathieu just got sent to penalty box.",
  "Next player : Thomas.",
]
`;

exports[`3rd scenario : player goes to penalty box then comes out 1`] = `
[
  "New player added : Mathieu at place 0, with 0 coin and not in penalty box.",
  "New player added : Thomas at place 0, with 0 coin and not in penalty box.",
  "Player Mathieu got 1 on their dice.",
  "Player Mathieu's new place is 1.",
  "Science Question 0",
  "Player Mathieu just got sent to penalty box.",
  "Next player : Thomas.",
  "Player Thomas got 2 on their dice.",
  "Player Thomas's new place is 2.",
  "Sports Question 0",
  "Player Thomas got one more coin. They have 1 coin(s) in their purse.",
  "Next player : Mathieu.",
  "Player Mathieu got 3 on their dice.",
  "Player Mathieu just got out of penalty box.",
  "Player Mathieu's new place is 4.",
  "Pop Question 0",
  "Player Mathieu got one more coin. They have 1 coin(s) in their purse.",
  "Next player : Thomas.",
]
`;

exports[`4th scenario : player goes to penalty box and stays there 1`] = `
[
  "New player added : Mathieu at place 0, with 0 coin and not in penalty box.",
  "New player added : Thomas at place 0, with 0 coin and not in penalty box.",
  "Player Mathieu got 1 on their dice.",
  "Player Mathieu's new place is 1.",
  "Science Question 0",
  "Player Mathieu just got sent to penalty box.",
  "Next player : Thomas.",
  "Player Thomas got 2 on their dice.",
  "Player Thomas's new place is 2.",
  "Sports Question 0",
  "Player Thomas got one more coin. They have 1 coin(s) in their purse.",
  "Next player : Mathieu.",
  "Player Mathieu got 4 on their dice.",
  "Player Mathieu stays in penalty box.",
]
`;

exports[`5th scenario : player goes to penalty box, comes out but goes back into it 1`] = `
[
  "New player added : Mathieu at place 0, with 0 coin and not in penalty box.",
  "New player added : Thomas at place 0, with 0 coin and not in penalty box.",
  "Player Mathieu got 1 on their dice.",
  "Player Mathieu's new place is 1.",
  "Science Question 0",
  "Player Mathieu just got sent to penalty box.",
  "Next player : Thomas.",
  "Player Thomas got 2 on their dice.",
  "Player Thomas's new place is 2.",
  "Sports Question 0",
  "Player Thomas got one more coin. They have 1 coin(s) in their purse.",
  "Next player : Mathieu.",
  "Player Mathieu got 3 on their dice.",
  "Player Mathieu just got out of penalty box.",
  "Player Mathieu's new place is 4.",
  "Pop Question 0",
  "Player Mathieu just got sent to penalty box.",
  "Next player : Thomas.",
]
`;

exports[`6th scenario : one player circles the board 1`] = `
[
  "New player added : Mathieu at place 0, with 0 coin and not in penalty box.",
  "New player added : Thomas at place 0, with 0 coin and not in penalty box.",
  "New player added : Clément at place 0, with 0 coin and not in penalty box.",
  "Player Mathieu got 6 on their dice.",
  "Player Mathieu's new place is 6.",
  "Sports Question 0",
  "Player Mathieu got one more coin. They have 1 coin(s) in their purse.",
  "Next player : Thomas.",
  "Player Thomas got 1 on their dice.",
  "Player Thomas's new place is 1.",
  "Science Question 0",
  "Player Thomas got one more coin. They have 1 coin(s) in their purse.",
  "Next player : Clément.",
  "Player Clément got 2 on their dice.",
  "Player Clément's new place is 2.",
  "Sports Question 1",
  "Player Clément got one more coin. They have 1 coin(s) in their purse.",
  "Next player : Mathieu.",
  "Player Mathieu got 6 on their dice.",
  "Player Mathieu has circled the board.",
  "Player Mathieu's new place is 0.",
  "Pop Question 0",
  "Player Mathieu got one more coin. They have 2 coin(s) in their purse.",
  "Next player : Thomas.",
]
`;

exports[`7th scenario : one player wins 1`] = `
[
  "New player added : Mathieu at place 0, with 0 coin and not in penalty box.",
  "New player added : Thomas at place 0, with 0 coin and not in penalty box.",
  "Player Mathieu got 1 on their dice.",
  "Player Mathieu's new place is 1.",
  "Science Question 0",
  "Player Mathieu got one more coin. They have 1 coin(s) in their purse.",
  "Next player : Thomas.",
  "Player Thomas got 1 on their dice.",
  "Player Thomas's new place is 1.",
  "Science Question 1",
  "Player Thomas got one more coin. They have 1 coin(s) in their purse.",
  "Next player : Mathieu.",
  "Player Mathieu got 1 on their dice.",
  "Player Mathieu's new place is 2.",
  "Sports Question 0",
  "Player Mathieu got one more coin. They have 2 coin(s) in their purse.",
  "Next player : Thomas.",
  "Player Thomas got 1 on their dice.",
  "Player Thomas's new place is 2.",
  "Sports Question 1",
  "Player Thomas got one more coin. They have 2 coin(s) in their purse.",
  "Next player : Mathieu.",
  "Player Mathieu got 1 on their dice.",
  "Player Mathieu's new place is 3.",
  "Rock Question 0",
  "Player Mathieu got one more coin. They have 3 coin(s) in their purse.",
  "Next player : Thomas.",
  "Player Thomas got 1 on their dice.",
  "Player Thomas's new place is 3.",
  "Rock Question 1",
  "Player Thomas got one more coin. They have 3 coin(s) in their purse.",
  "Next player : Mathieu.",
  "Player Mathieu got 1 on their dice.",
  "Player Mathieu's new place is 4.",
  "Pop Question 0",
  "Player Mathieu got one more coin. They have 4 coin(s) in their purse.",
  "Next player : Thomas.",
  "Player Thomas got 1 on their dice.",
  "Player Thomas's new place is 4.",
  "Pop Question 1",
  "Player Thomas got one more coin. They have 4 coin(s) in their purse.",
  "Next player : Mathieu.",
  "Player Mathieu got 1 on their dice.",
  "Player Mathieu's new place is 5.",
  "Science Question 2",
  "Player Mathieu got one more coin. They have 5 coin(s) in their purse.",
  "Next player : Thomas.",
  "Player Thomas got 1 on their dice.",
  "Player Thomas's new place is 5.",
  "Science Question 3",
  "Player Thomas got one more coin. They have 5 coin(s) in their purse.",
  "Next player : Mathieu.",
  "Player Mathieu got 1 on their dice.",
  "Player Mathieu's new place is 6.",
  "Sports Question 2",
  "Player Mathieu got one more coin. They have 6 coin(s) in their purse.",
  "Player Mathieu has won.",
  "Next player : Thomas.",
]
`;

exports[`My first scenario 1`] = `
[
  "New player added : Mathieu at place 0, with 0 coin and not in penalty box.",
  "New player added : Thomas at place 0, with 0 coin and not in penalty box.",
  "New player added : Clément at place 0, with 0 coin and not in penalty box.",
  "Player Mathieu got 1 on their dice.",
  "Player Mathieu's new place is 1.",
  "Science Question 0",
  "Player Mathieu got one more coin. They have 1 coin(s) in their purse.",
  "Next player : Thomas.",
  "Player Thomas got 1 on their dice.",
  "Player Thomas's new place is 1.",
  "Science Question 1",
  "Player Thomas got one more coin. They have 1 coin(s) in their purse.",
  "Next player : Clément.",
  "Player Clément got 1 on their dice.",
  "Player Clément's new place is 1.",
  "Science Question 2",
  "Player Clément got one more coin. They have 1 coin(s) in their purse.",
  "Next player : Mathieu.",
  "Player Mathieu got 1 on their dice.",
  "Player Mathieu's new place is 2.",
  "Sports Question 0",
  "Player Mathieu got one more coin. They have 2 coin(s) in their purse.",
  "Next player : Thomas.",
  "Player Thomas got 1 on their dice.",
  "Player Thomas's new place is 2.",
  "Sports Question 1",
  "Player Thomas got one more coin. They have 2 coin(s) in their purse.",
  "Next player : Clément.",
  "Player Clément got 1 on their dice.",
  "Player Clément's new place is 2.",
  "Sports Question 2",
  "Player Clément got one more coin. They have 2 coin(s) in their purse.",
  "Next player : Mathieu.",
  "Player Mathieu got 1 on their dice.",
  "Player Mathieu's new place is 3.",
  "Rock Question 0",
  "Player Mathieu got one more coin. They have 3 coin(s) in their purse.",
  "Next player : Thomas.",
  "Player Thomas got 1 on their dice.",
  "Player Thomas's new place is 3.",
  "Rock Question 1",
  "Player Thomas got one more coin. They have 3 coin(s) in their purse.",
  "Next player : Clément.",
]
`;
